nixos/doc/manual/release-notes/rl-2305.section.md

```diff
@@ -198,6 +198,11 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 - The [services.wordpress.sites.&lt;name&gt;.plugins](#opt-services.wordpress.sites._name_.plugins) and [services.wordpress.sites.&lt;name&gt;.themes](#opt-services.wordpress.sites._name_.themes) options have been converted from sets to attribute sets to allow for consumers to specify explicit install paths via attribute name.
 
+- [`services.nextcloud.database.createLocally`](#opt-services.nextcloud.database.createLocally) now uses socket authentication and is no longer compatible with password authentication.
+  - If you want the module to manage the database for you, unset [`services.nextcloud.config.dbpassFile`](#opt-services.nextcloud.config.dbpassFile) (and [`services.nextcloud.config.dbhost`](#opt-services.nextcloud.config.dbhost), if it's set).
+  - If your database is external, simply set [`services.nextcloud.database.createLocally`](#opt-services.nextcloud.database.createLocally) to `false`.
```

Comment:
As far as I can see, I'm sure 99,9% of people using `config.dbhost` also have `config.dbpassFile` set, like so:
```nix
{
  services.nextcloud = {
    enable = true;
    config.dbhost = "example.com";
    config.dbpassFile = "/run/secrets/foo";
  };
}
```

This will trigger the assertion, and then the user will know they will have to add `database.createLocally = false` to their config.